### ISSUE TITLE
error restarting workflow

### ISSUE THREAD

**Issue by andypohl** [pegasus-isi/dipa-workflow#4]
We're getting an error after
1. Setting up DAX and submitting.  
2. We run pegasus-analyzer wrapped around a condor_q
3. Workflow finishes, with errors.
4. We fix one of the DIPA scripts, presumably breaking the workflow.
5. We rerun script to make the DAX, and it errors.

(stay tuned for a better explanation)

**Comment by andypohl** [comment]
I was trying to move this to my fork, but I guess there's only one place for issues.

**Comment by AndrewJSchoen** [comment]
This appears to be an issue with pegasus-analyzer, where if we rerun DIPA, it dies. I will try to find the error message when I can recreate it.

On a related note, though. If we rerun a dax with modifications, does this by default figure out what needs to be rerun, or is there something setup-wise that we need to do in order to enable this feature?

**Comment by andypohl** [comment]
just checking, but did we use pegasus-run to re-run the workflow or pegasus-plan?  pegasus-run is the program to use for rescuing workflows.

**Comment by AndrewJSchoen** [comment]
Good question.
I did run pegasus-plan to rerun, so that could be an issue. However, a common occurrence will be that, for example, a new subject will be added such that pegasus-plan will need to be rerun, since technically the structure of the workflow will be modified. Is this the correct course of action?

**Comment by vahi** [comment]
Yes if you are changing the DAX then pegasus-plan needs to be run.

If you want to trigger data reuse, then you can remove the —force option to pegasus-plan and 
a) setup the replica catalog to have locations of intermediate output files from the previous run
OR
b) pass the the scratch directory from previous run as —input-dir option to pegasus-plan

Thanks
KAran
